### DISCUSSION TITLE
Voltage Link Connector

### DISCUSSION THREAD

**Discussion by randel0** [hatlabs/discussions#30] [General]
Is the Voltage Link Connector depicted on the schematics?

I'm trying to troubleshoot a stubborn problem, and have discovered that my project works as expected when I power it from the Voltage Link Connector on my workbench.

When I power the project from the NMEA bus, the input of my ADS1115 goes into self-oscillation, and is unusable.

Do you have any comments or suggestions?

The ADS115 is connected to the output of a Honeywell position sensor via a voltage divider.

Thanks!

-Randel

**Comment by randel0** [comment]
https://github.com/randel0/NEMA2K-Mast-Rotation-Compensator

**Comment by mairas** [comment]
(Assuming this is about SH-ESP32.)

The Voltage Link is on the schematics, but admittedly not very clearly - seems like it's called "Wire-link" there. The CAN side connector is J501 on page 2 of the [v2.0.1 schema](https://hatlabs.github.io/sh-esp32/pages/revisions/assets/SH-ESP32-2.0.1-schema.pdf), while the "main" board side connector is J204 on page 8.

What is your circuit setup like?
Do you have both positive and negative wires connected to your measurement?
If so, you're having a galvanic connection to the system being measured. Now, when you power the system through N2K, you're providing GND through two distinct circuits, potentially causing a ground loop. Ground loops are notorious for inducing noise in the system - for example, the "mains hum" in audio or PA systems is typically caused by them. It sounds like that's what happening.

One thing to try is to disconnect the negative lead from the measurement to the ADS1115. That would break the ground loop, although you might still have noise through the positive lead.
NMEA 2000 standard mandates isolated interfaces, so with the ADS1115, you should power the system using the dedicated power connector. Admittedly, you still might have a GND loop via the power connection and the ADS1115 measurement, but at least that wouldn't involve your navigation systems...

Electrical noise can be a pain to figure out and fix - hope this helps you somewhat.

**Comment by randel0** [comment]
Thank you for the response, and I agree that it is probably a ground loop problem.

To be clear, the Voltage Link jumper marked GND is actually CAN_GND, and this is isolated from the other pads marked GND with U502?

Are the GND pads in the upper left corner an isolated GND?

**Reply by mairas** [reply]
Not sure if I follow. :-)

CAN_GND is isolated from GND on the rest of the board. All of the board's functions except the CAN and OPTO I/O subsystems use GND as their zero reference. CAN and Opto I/O are galvanically isolated and therefore have their own reference grounds. If you want to power your board via the NMEA 2000 connection, then you need to give up the galvanic isolation and connect GND and CAN_GND together, either by splitting the power wire to both connectors, or by using the Voltage Link.

Another way to provide power while retaining the isolation would be to use isolated DC-DC converters, but they take a lot of space and cost quite a bit.
